Mesh/d4est_mesh.c

```diff
@@ -1283,7 +1283,7 @@ d4est_mesh_compute_l2_norm_sqr
 {
   int k = 0;
   double* Mvec = P4EST_ALLOC(double, local_nodes);
```

Comment:
Could you comment on your implementation of the L_2 norm a bit (perhaps in the code)? When I just computed it from the square sum of the field value errors and normalised by the square of the number of nodes, I obtained a slightly different number. I guess this is because you perform some sort of weighting here?

Reply:
The square sum of the field - field_compare values won't give the same value because this is the integral L2 norm:
![image](https://user-images.githubusercontent.com/6779417/36004482-f7cf57c6-0d00-11e8-8324-a992b9c4e8b3.png)
Where J_i is the jacobian and w_i is the quadrature weights

Reply:
I'll add a doxygen comment for the function into the master branch. I need to comment a lot more and get some good doxygen documentation up and going. It's on the todo list.